File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* list1, ListNode* list2) {
           if(list1 == NULL) return list2;
           if(list2 == NULL) return list1;
           ListNode* res = NULL;
          if(list1->val < list2->val){
              res = list1;
              list1 = list1->next;
          }
        else{
            res = list2;
            list2 = list2->next;
        }
        ListNode* dummy = res;
        while(list1 != NULL && list2 != NULL){
            ListNode* temp = NULL;
            if(list1->val < list2->val){
                temp = list1;
                list1 = list1->next;
            }
            else{
                temp = list2;
                list2 = list2->next;
            }
            dummy->next = temp;
            dummy = temp;
        }
        if(list1 != NULL) dummy->next = list1;
        else {
            dummy->next = list2;
        }
        return res;
    }
};

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        if(list1 == null)return list2;
        if(list2 == null)return list1;
        
        ListNode s = null;
        if(list1.val <= list2.val){
            s = list1;
            list1 = list1.next;
        }
        else{
            s = list2;
            list2 = list2.next;
        }
        
        ListNode dummy = s;
        while(list1 != null && list2 != null){
            if(list1.val <= list2.val){
                dummy.next = list1;
                dummy = dummy.next;
                list1 = list1.next;
            }
            else{
                dummy.next = list2;
                dummy = dummy.next;
                list2 = list2.next;
            }
        }
        if(list1 == null){
            dummy.next = list2;
        }
        else{
            dummy.next = list1;
        }
        return s;
    }
}